### PR TITLE
Added service tests

### LINKED ISSUE #4: service tests
Add integration tests for the [service/service](https://github.com/cryptic-game/cryptic-service/wiki/API-Reference-[experimental]#services) endpoints

### PR DESCRIPTION
**Pull request by T2brozz**
**Description**
Added tests for service endpoints.

**Issue**
Closes #4

### CODE REVIEW

File: tests/test_service.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        cls.client.login("super", super_password)
+
+    @classmethod
+    def tearDownClass(cls: "TestService"):
+        cls.client.close()
+
+    def test_public_info_successful(self):
+        device_uuid = setup_device()[0]
+        service_uuid = create_service(device_uuid)[0]
+        actual = self.client.ms("service", ["public_info"], device_uuid=device_uuid, service_uuid=service_uuid)
```
Comment (Defelo) [inline tests/test_service.py:64]: Please leave an empty line between setup and assertions to improve readability. (applies to other test cases as well)

File: tests/test_service.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+def create_service(device=uuid(), name="telnet", n=1, part_owner=None, owner=super_uuid, clear_service=True):
+    if clear_service:
+        clear_services()
+
+    service_uuids = []
+    for i in range(n):
+        service_uuids.append(uuid())
+        execute(
+            "INSERT INTO service_service(uuid, device, owner, name, running,running_port, part_owner)"
+            + "VALUES (%s,%s,%s,%s,%s,%s,%s)",
```
Comment (Defelo) [inline tests/test_service.py:33]:
```suggestion
            "INSERT INTO service_service (uuid, device, owner, name, running, running_port, part_owner) "
            "VALUES (%s,%s,%s,%s,%s,%s,%s)",
```

File: tests/test_service.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+)
+
+from database import execute
+from testcase import TestCase
+from tests.test_device import setup_device
+from tests.test_server import setup_account, super_password, super_uuid
+from util import get_client, uuid
+
+
+def create_service(device=uuid(), name="telnet", n=1, part_owner=None, owner=super_uuid, clear_service=True):
```
Comment (Defelo) [inline tests/test_service.py:24]: Don't call other functions in function definitions. As this is a required argument, remove the default value instead.

File: tests/test_service.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+        actual = self.client.ms("service", ["private_info"], device_uuid=device_uuid, service_uuid=service_uuid)
+        self.assert_dict_with_keys(
+            actual, ["running", "owner", "running_port", "name", "uuid", "device", "speed", "part_owner"]
+        )
+        self.assertTrue(actual["running"])
+        self.assertEqual(actual["owner"], super_uuid)
+        self.assertEqual(actual["running_port"], 1337)
+        self.assertEqual(actual["name"], "telnet")
+        self.assertEqual(actual["uuid"], service_uuid)
+        self.assertEqual(actual["device"], device_uuid)
```
Comment (Defelo) [inline tests/test_service.py:149]: The first parameter of the `assertEqual` function should be the expected value and the second one the actual value. (applies to other test cases as well)

File: tests/test_service.py [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+    def test_public_info_successful(self):
+        device_uuid = setup_device()[0]
+        service_uuid = create_service(device_uuid)[0]
+        actual = self.client.ms("service", ["public_info"], device_uuid=device_uuid, service_uuid=service_uuid)
+        self.assert_dict_with_keys(actual, ["running_port", "name", "uuid", "device"])
+        self.assertEqual(actual["device"], device_uuid)
+        self.assertEqual(actual["uuid"], service_uuid)
+        self.assertEqual(actual["running_port"], 1337)
+        self.assertEqual(actual["name"], "telnet")
```
Comment (Defelo) [inline tests/test_service.py:69]: Please define an `expected` variable and use `self.assertEqual(expected, actual)` instead. (applies to other test cases as well)

**Review by Defelo — CHANGES_REQUESTED** [review]

**Review by Defelo — APPROVED** [review]

**Review by MarcelCoding — APPROVED** [review]
Maby Merge 78a28e869d988da764ec0b00028753e205a455b6 and 44e49d833d0e5a9c72257d42ddb43ed519b4fd04

**Pull request merged**